android/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java

```diff
@@ -37,42 +41,52 @@ function capacitorSystemBarsCheckMetaViewport() {
             const metaContent = meta[meta.length - 1].content;
             return metaContent.includes("viewport-fit=cover");
         }
-
         capacitorSystemBarsCheckMetaViewport();
         """;
 
+    private boolean insetHandlingEnabled = true;
+    private boolean hasViewportCover = false;
+
     @Override
     public void load() {
+        getBridge().getWebView().addJavascriptInterface(this, "CapacitorSystemBarsAndroidInterface");
         super.load();
+
         initSystemBars();
     }
 
-    private boolean hasFixedWebView() {
-        PackageInfo packageInfo = WebViewCompat.getCurrentWebViewPackage(bridge.getContext());
-        Pattern pattern = Pattern.compile("(\\d+)");
-        Matcher matcher = pattern.matcher(packageInfo.versionName);
-
-        if (!matcher.find()) {
-            return false;
-        }
-
-        String majorVersionStr = matcher.group(0);
-        int majorVersion = Integer.parseInt(majorVersionStr);
+    @Override
+    protected void handleOnStart() {
+        super.handleOnStart();
+
+        this.getBridge().addWebViewListener(
+            new WebViewListener() {
+                @Override
+                public void onPageCommitVisible(WebView view, String url) {
+                    super.onPageCommitVisible(view, url);
+                    getBridge().getWebView().requestApplyInsets();
```

Comment:
How does this relate to the `onDOMReady` code? With `onDOMReady` already in place, this code doesn't seem to be necessary? Especially since `viewportMetaJSFunction` is not executed here (again)